From ad6ab3873efb15b5cfa034689e0f886c9d3d0f4d Mon Sep 17 00:00:00 2001
From: Maggie <yiranwang@google.com>
Date: Wed, 24 Jan 2018 11:01:20 -0800
Subject: [PATCH 072/306] Fix location INTERACT_ACROSS_USERS bug

Use UserHandle.getCallingUserId in LocationManager.isProviderEnabled

Bug: 72430849
Test: Manual
Change-Id: I1ddfc4fd308889b1f2247df038d22e2f7b08da74
(cherry picked from commit 9ebcf05596ec242a601924df46a951459ca59a72)
---
 .../java/com/android/server/LocationManagerService.java     | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/services/core/java/com/android/server/LocationManagerService.java b/services/core/java/com/android/server/LocationManagerService.java
index 5a4f7cab3ed..9aa588fc823 100644
--- a/services/core/java/com/android/server/LocationManagerService.java
+++ b/services/core/java/com/android/server/LocationManagerService.java
@@ -2570,7 +2570,7 @@ public class LocationManagerService extends ILocationManager.Stub {
         // Check INTERACT_ACROSS_USERS permission if userId is not current user id.
         checkInteractAcrossUsersPermission(userId);
 
-        // enable all location providers
+        // Enable or disable all location providers
         synchronized (mLock) {
             for(String provider : getAllProviders()) {
                 setProviderEnabledForUser(provider, enabled, userId);
@@ -2586,10 +2586,10 @@ public class LocationManagerService extends ILocationManager.Stub {
      */
     @Override
     public boolean isLocationEnabledForUser(int userId) {
-
         // Check INTERACT_ACROSS_USERS permission if userId is not current user id.
         checkInteractAcrossUsersPermission(userId);
 
+        // If at least one location provider is enabled, return true
         synchronized (mLock) {
             for (String provider : getAllProviders()) {
                 if (isProviderEnabledForUser(provider, userId)) {
@@ -2602,7 +2602,7 @@ public class LocationManagerService extends ILocationManager.Stub {
 
     @Override
     public boolean isProviderEnabled(String provider) {
-        return isProviderEnabledForUser(provider, UserHandle.myUserId());
+        return isProviderEnabledForUser(provider, UserHandle.getCallingUserId());
     }
 
     /**
-- 
2.17.1

